File: src/main/java/refactown/cleancode/c08boundary/java/GsonJsonParser.java
package refactown.cleancode.c08boundary.java;

import com.google.gson.Gson;

import refactown.cleancode.c07exceptions.java.JsonException;

public class GsonJsonParser implements JsonParser {

	private Gson gson = new Gson();

	@Override
	public Object fromJson(String json, Class<?> clazz) {
		try {
			return gson.fromJson(json, clazz);
		} catch (RuntimeException e) {
			throw new JsonException("Parsing error. Json:" + json, e);
		}
	}

	@Override
	public String toJson(Object obj) {
		try {
			return gson.toJson(obj);
		} catch (RuntimeException e) {
			throw new JsonException("Serializing error. Object:" + obj, e);
		}

	}

}

File: src/main/kotlin/refactown/cleancode/c08boundary/GsonJsonParser.kt
package refactown.cleancode.c08boundary

import com.google.gson.Gson
import refactown.cleancode.c07exceptions.JsonException

class GsonJsonParser: JsonParser{

    private val gson = Gson()

    override fun toJson(obj: Any): String {
        try {
            return gson.toJson(obj)
        } catch (e: Exception){
            throw JsonException("Serialize error. Object:$obj", e)
        }
    }

    override fun fromJson(json: String, clazz: Class<*>): Any {
        try {
            return gson.fromJson(json, clazz)
        } catch (e: Exception){
            throw JsonException("Parsing error. Json text:$json", e)
        }
    }

}